packages/cli/src/checks/__tests__/EXTERNAL_MISMATCH.ts

```diff
@@ -224,9 +224,38 @@ it("should not error if the value is not a valid semver range", () => {
   };
   ws.set("pkg-2", pkg2);
 
-  let errors = internalMismatch.validate(pkg2, ws, rootWorkspace, {});
+  let errors = externalMismatch.validate(pkg2, ws, rootWorkspace, {});
   expect(errors.length).toEqual(0);
 
-  errors = internalMismatch.validate(ws.get("pkg-1")!, ws, rootWorkspace, {});
+  errors = externalMismatch.validate(ws.get("pkg-1")!, ws, rootWorkspace, {});
+  expect(errors.length).toEqual(0);
+});
+
```

Comment:
We should add a test case to validate which version is used when adding an unsupported version in the case where 2 or more allowed versions exist.
e.g.

```
'allowedDependencyVersions' {
     'foo': [
         '^1.0.0',
         '^2.0.0'
      ]
}
```
`yarn add foo@^3.0.0` ... does it clamp to the lowest version number, the highest one, or the most commonly used between the allowed versions?

Reply:
yep, added a test case, should be the most commonly used one.